ams/server/src/main/java/com/netease/arctic/server/persistence/SqlSessionFactoryProvider.java

```diff
@@ -107,32 +109,47 @@ public void init(Configurations config) {
     createTablesIfNeed(config);
   }
 
-  // create tables for derby database type
+  /**
+   * create tables for database
+   *
+   * @param config
+   */
   private void createTablesIfNeed(Configurations config) {
-    if (ArcticManagementConf.DB_TYPE_DERBY.equals(config.getString(ArcticManagementConf.DB_TYPE))) {
-      try (SqlSession sqlSession = get().openSession(true)) {
-        try (Connection connection = sqlSession.getConnection()) {
-          try (Statement statement = connection.createStatement()) {
-            String query = "SELECT 1 FROM SYS.SYSTABLES WHERE TABLENAME = 'CATALOG_METADATA'";
-            try (ResultSet rs = statement.executeQuery(query)) {
-              if (!rs.next()) {
-                ScriptRunner runner = new ScriptRunner(connection);
-                runner.runScript(new InputStreamReader(new FileInputStream(getDerbyInitSqlScriptPath()),
-                    StandardCharsets.UTF_8));
-              }
-            }
-          }
-        }
-      } catch (Exception e) {
-        throw new IllegalStateException("Create derby tables failed", e);
+    String dbTypeConfig = config.getString(ArcticManagementConf.DB_TYPE);
+    String query = "";
+
+    try (SqlSession sqlSession = get().openSession(true);
+         Connection connection = sqlSession.getConnection();
+         Statement statement = connection.createStatement();) {
+      if (ArcticManagementConf.DB_TYPE_DERBY.equals(dbTypeConfig)) {
+        query = "SELECT 1 FROM SYS.SYSTABLES WHERE TABLENAME = 'CATALOG_METADATA'";
+      } else if (ArcticManagementConf.DB_TYPE_MYSQL.equals(dbTypeConfig)) {
+        query = String.format(
+            "SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = '%s' AND table_name = '%s'",
```

Comment:
I tested locally with MySQL, but it didn't work.

I think we should change the `SELECT COUNT(*) FROM` to `SELECT * FROM`. @Kyofin

Reply:
Maybe `SELECT 1` is better, keepping the same as Derby.